src/data/repositories/ProjectD2Repository.ts

```diff
@@ -21,10 +21,52 @@ export class ProjectD2Repository implements ProjectRepository {
         this.d2ApiConfig = new D2ApiConfig(this.api);
     }
 
+    getList(): FutureData<Project[]> {
+        return this.getCategories().flatMap(categories => {
+            return apiToFuture(
+                this.api.models.categoryOptions.get({
+                    fields: {
+                        id: true,
+                        displayName: true,
+                        startDate: true,
+                        endDate: true,
+                        lastUpdated: true,
+                    },
+                    filter: { "categories.code": { eq: categories.project.code } },
+                    order: "displayName:asc",
+                    paging: false,
+                })
+            ).map(d2Response => {
+                return d2Response.objects.map((d2CategoryOption): Project => {
+                    return Project.build({
+                        dataSets: [],
+                        id: d2CategoryOption.id,
+                        name: d2CategoryOption.displayName,
+                        lastUpdated: d2CategoryOption.lastUpdated,
+                        isOpen: this.isProjectOpen(
+                            d2CategoryOption.startDate,
+                            d2CategoryOption.endDate
+                        ),
+                    });
+                });
+            });
+        });
+    }
+
     getAll(): FutureData<Project[]> {
         return this.getAllProjects(1, []);
     }
 
+    private isProjectOpen(date1: ISODateString, date2: ISODateString): boolean {
+        if (!date1 || !date2) return false;
```

Comment:
TS-smell. the type says we do have a ISODateString, but we need to check if empty. So probably the type is `Maybe<IsoDateString>`. I see this comes from a `d2CategoryOption.startDate`. This, I guess, is a shortcoming of d2-api, you ask for a field but this field may not be set so it's empty in the response. Until we have support for that in d2-api, cast explicitly as soon as possible.